src/lib.rs

```diff
@@ -53,6 +55,79 @@ create_exception!(johnnycanencrypt, SameKeyError, PyException);
 // Error in selecting OpenPGP applet in the card
 create_exception!(johnnycanencrypt, CardError, PyException);
 
+#[pyfunction]
+#[text_signature = "(certdata, uid, pass)"]
+pub fn revoke_uid_in_cert(
+    py: Python,
+    certdata: Vec<u8>,
+    uid: Vec<u8>,
+    pass: String,
+) -> PyResult<PyObject> {
+    // This is where we will store all the signing keys
+    let mut cert = openpgp::Cert::from_bytes(&certdata).unwrap();
+
+    // To find the secret keypair
+    let mut keypair = match cert
+        .primary_key()
+        .key()
+        .clone()
+        .parts_into_secret()
+        .unwrap()
+        .secret()
+        .is_encrypted()
+    {
+        true => {
+            // When the secret is encrypted with a password
+            cert.primary_key()
+                .key()
+                .clone()
+                .parts_into_secret()
+                .unwrap()
+                .decrypt_secret(&openpgp::crypto::Password::from(pass))
+                .unwrap()
+                .into_keypair()
+                .unwrap()
+        }
+        false => {
+            // When the secret is not encrypted
+            cert.primary_key()
+                .key()
+                .clone()
+                .parts_into_secret()
+                .unwrap()
+                .into_keypair()
+                .unwrap()
+        }
+    };
+
+    // now let us find the user id from the cert.
+    for ua in cert.clone().userids() {
+        let value = ua.value().to_vec();
+        if value == uid {
+            // We found the uid which we can now revoke
+            println!("we have a match");           let sig = UserIDRevocationBuilder::new()
```

Comment:
Indentation?

Reply:
Extra debugging code, has to be removed and formatted.